### ISSUE TITLE
Alternators

### ISSUE THREAD

**Issue by NathanKell** [KSP-RO/RealismOverhaul#41]
How should alternators be handled? Were (are) the avionics and the payload always battery powered, or did (do) pump-fed engines hook into the main circuits with their generators?
I would guess with pressure-fed engines (which lack generators to run the pumps) no alternator should be simulated, but for pump-fed engines...?

**Comment by RedAV8R** [comment]
From what I can tell besides actual sensors and what not that are obviously connected to the brains of a launch vehicle the actual power is also supplied from that vehicle or stage, not the engine. That I know of pump fed engines do not have an electrical generator to power pumps. If anything they would have a starter motor, fed by battery power, to start the pump initially until it could be ignited. Really, it's not so much different than an aircraft. Battery power starts a starter/generator, which spins the compressor and/or turbine, which when fuel is introduced and ignited, creates a self-sustaining reaction as long as fuel continues to flow. Now obviously this is where things differ, in that an aircraft starter/generator changes function from starter to generator to resupply the aircraft. From what I can tell, the RS-25 had it's engine controllers on the engine, they were hooked up to the vehicle through it's own bus. Likely again powered by the vehicle not the engine. But what do I know. No Holiday Inn Express for me last night.

In conclusion I recommend that rocket engines do NOT have a generator, however air breathing engines do.

Perhaps we need to make a 'standards' policy file for configs for others to read and conform to?

**Comment by NathanKell** [comment]
That makes sense to me, on both counts. :)

**Comment by kiwivogel** [comment]
SSME engine controllers were powered by the shuttle's fuel cells. With 2 redundant controllers per engine on separate busses (and backup switchover capabilities).

**Comment by stratochief66** [comment]
@NathanKell close?
